common/src/main/java/com/wynntils/models/inventory/InventoryModel.java

```diff
@@ -83,6 +85,25 @@ public List<ItemStack> getEquippedItems() {
         return returnable.stream().filter(itemStack -> !itemStack.isEmpty()).toList();
     }
 
+    /**
+     * @return True if the player has any item with AUTO_CASTER_MAJOR_IDS major id(s)
+     */
+    public boolean hasAutoCasterItem() {
+        for (ItemStack item : getEquippedItems()) {
+            Optional<GearItem> gearItemOpt = Models.Item.asWynnItem(item, GearItem.class);
+            if (gearItemOpt.isEmpty()) continue;
+
+            GearItem gearItem = gearItemOpt.get();
+            if (gearItem.getItemInfo().fixedStats().majorIds().stream()
+                    .anyMatch(majorId -> AUTO_CASTER_MAJOR_IDS.contains(majorId.name()))) {
+                return true;
+            }
+        }
+
+        return Models.Raid.getRaidMajorIds(McUtils.mc().getUser().getName()).stream()
+                .anyMatch(AUTO_CASTER_MAJOR_IDS::contains);
+    }
+
```

Comment:
This may not exactly be inventory model code but I'm fine with keeping it here until we need to use it more often.

Reply:
Kind of agreed, just because it checks for raid major ids. Maybe someone can move to CharacterStats or StatusEffects or Spell and rename next time it's used.